### DISCUSSION TITLE
Gap8 bootloader (for the AIdeck 1.0 or AIdeck 1.1 Rev A-C)

### DISCUSSION THREAD

**Discussion by mjm41** [bitcraze/discussions#707] [Q&A] [answered]
My configuration:
1. Microsoft Surface 3 laptop
2. Bitcraze VM (Ubuntu-based)
3. [Crazyflie 2.1](https://www.bitcraze.io/products/crazyflie-2-1/) < firmware updated
4. [AI Deck](https://www.bitcraze.io/products/ai-deck/) < version C1
5. [Olimex ARM-USB-TINY-H JTAG](https://www.olimex.com/Products/ARM/JTAG/ARM-USB-TINY-H/)
6. [Olimex ARM-JTAG-20-10](https://www.olimex.com/Products/ARM/JTAG/ARM-JTAG-20-10/)
7. [Olimex CABLE-USB-A-B-1.8M](https://www.olimex.com/Products/Components/Cables/CABLE-USB-A-B-1.8M/)

I have followed the [Getting started with the AI deck](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/) tutorial and am stuck figuring out how to ensure proper connection of the JTAG device for the Gap8 bootloader step. The ARM-USB-TINY-JTAG and ARM-JTAG-20-10 both have female ends. Do I just find a 20-pin connector like this: [Connector Header Through Hole 20 position 0.100" (2.54mm)](https://www.digikey.com/en/products/detail/molex/0878342019/3303513?utm_adgroup=Connectors%2C%20Interconnects&utm_source=google&utm_medium=cpc&utm_campaign=Shopping_Supplier_Molex_0900_Co-op&utm_term=&utm_content=Connectors%2C%20Interconnects&gclid=Cj0KCQjw0tKiBhC6ARIsAAOXutnkrdcfRgSeVHUs8vdtjZwUk94DRlWGjOUyqc9WHgQdKL5y_FF6qIIaAtF0EALw_wcB)?

Additionally, can anyone explain the more detailed process for this step? For example, should it work if I plug everything together (with crazyflie off), ensure USB is passing through into the VM, then clone repo and cd into gap8 directory, then before docker command, turn crazyflie on, let it open into bootloader mode and finally run the docker command?

Thank you.

**Comment by mjm41** [comment, marked as answer]
I suppose the problem is solved. After reading the JTAG programmer manual I read that it has a male end. Perplexed I pulled on the cord that came attached and found that it comes off. I attached the 20-10 pin adaptor and ran the docker code with success.

**Reply by knmcguire** [reply]
Great! thanks for updating us on this
